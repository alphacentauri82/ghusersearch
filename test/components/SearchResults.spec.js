import React from 'react';
import renderer from 'react-test-renderer';
import {StyleSheetTestUtils} from 'aphrodite';
import {
  StaticRouter as Router,
} from 'react-router-dom';

import SearchResults from 'components/SearchResults';

function createComponent(props) {
  return renderer.create(
    <Router context={{}}>
      <SearchResults {...props} />
    </Router>
  );
}

describe('Component: SearchResults', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  describe('when given a set of results and entities', () => {
    it('should render them with a message', () => {
      const props = {
        entities: {
          1: {login: 'foo', avatar_url: 'avatar.jpg', id: 1},
          2: {login: 'foobar', avatar_url: 'avatar.jpg', id: 2},
        },
        ids: [1, 2],
        searchTerm: 'foo',
        total: 2,
      };
      const component = createComponent(props);
      expect(component).toMatchSnapshot();
    });
  });

  describe('when given a one result', () => {
    it('should render a correctly worded message', () => {
      const props = {
        entities: {
          1: {login: 'foo', avatar_url: 'avatar.jpg', id: 1},
        },
        ids: [1],
        searchTerm: 'foo',
        total: 1,
      };
      const component = createComponent(props);
      expect(component).toMatchSnapshot();
    });
  });

  describe('when the results are 0', () => {
    it('should not render a message', () => {
      const props = {
        entities: {},
        ids: [],
        searchTerm: 'foo',
        total: 0,
      };
      const component = createComponent(props);
      expect(component).toMatchSnapshot();
    });
  });

});
