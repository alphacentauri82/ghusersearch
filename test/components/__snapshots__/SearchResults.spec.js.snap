// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SearchResults when given a one result should render a correctly worded message 1`] = `
<ul
  className="SearchResults_1uxrx9c u-sm-flex u-sm-flexWrap"
>
  <li
    className="SearchResults_item_1mclqe7"
  >
    <a
      className="User_abw475 u-flex"
      href="/foo"
      onClick={[Function]}
    >
      <div
        className="User_wrapAvatar_3afm6b"
      >
        <div
          className="Image_e296pg"
          style={
            Object {
              "minHeight": 60,
            }
          }
        >
          <img
            alt="foo's avatar"
            className="Avatar_1pcpsid u-imgResponsive"
            onLoad={[Function]}
            src="avatar.jpg&s=180"
            width={90}
          />
          <svg />
        </div>
      </div>
      <p
        className="User_username_rnfkwd"
      >
        foo
      </p>
    </a>
  </li>
</ul>
`;

exports[`Component: SearchResults when given a set of results and entities should render them with a message 1`] = `
<ul
  className="SearchResults_1uxrx9c u-sm-flex u-sm-flexWrap"
>
  <li
    className="SearchResults_item_1mclqe7"
  >
    <a
      className="User_abw475 u-flex"
      href="/foo"
      onClick={[Function]}
    >
      <div
        className="User_wrapAvatar_3afm6b"
      >
        <div
          className="Image_e296pg"
          style={
            Object {
              "minHeight": 60,
            }
          }
        >
          <img
            alt="foo's avatar"
            className="Avatar_1pcpsid u-imgResponsive"
            onLoad={[Function]}
            src="avatar.jpg&s=180"
            width={90}
          />
          <svg />
        </div>
      </div>
      <p
        className="User_username_rnfkwd"
      >
        foo
      </p>
    </a>
  </li>
  <li
    className="SearchResults_item_1mclqe7"
  >
    <a
      className="User_abw475 u-flex"
      href="/foobar"
      onClick={[Function]}
    >
      <div
        className="User_wrapAvatar_3afm6b"
      >
        <div
          className="Image_e296pg"
          style={
            Object {
              "minHeight": 60,
            }
          }
        >
          <img
            alt="foobar's avatar"
            className="Avatar_1pcpsid u-imgResponsive"
            onLoad={[Function]}
            src="avatar.jpg&s=180"
            width={90}
          />
          <svg />
        </div>
      </div>
      <p
        className="User_username_rnfkwd"
      >
        foobar
      </p>
    </a>
  </li>
</ul>
`;

exports[`Component: SearchResults when the results are 0 should not render a message 1`] = `
<ul
  className="SearchResults_1uxrx9c u-sm-flex u-sm-flexWrap"
/>
`;
